independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/ClientRequestContextImpl.java

```diff
@@ -56,7 +56,11 @@ public ClientRequestContextImpl(RestClientRequestContext restClientRequestContex
 
         // Capture or create a duplicated context, and store it.
         Context current = client.vertx.getOrCreateContext();
-        this.context = VertxContext.getOrCreateDuplicatedContext(current);
+        if (VertxContext.isDuplicatedContext(current)) {
+            this.context = VertxContext.createNewDuplicatedContext();
```

Comment:
The problem here is that we check that `current` is a duplicated context, but create a new one from Vertx.currentContext(), which may not be `current`

Reply:
Yes, that is why we need the API that creates a new duplicated context from a specified context. Right now that is the only one we have.

Reply:
or merge the PR I just opened on smallrye commons :-D

Reply:
Thanks :)